src/Core/include/OpenLoco/Core/Numerics.hpp

```diff
@@ -27,4 +27,16 @@ namespace OpenLoco::Numerics
     {
         return ((val >> rotation) | (val << (4 - rotation))) & 0xF;
     }
+
+    template<typename T>
+    constexpr T alignDown(T value, size_t alignment)
+    {
+        return value & ~(alignment - 1);
```

Comment:
I find this much harder to understand than `& ~0 <<alignment`

Reply:
It is however less efficient in terms of compiled output code.
```
alignDown32_PR(unsigned int, unsigned long):
        mov     eax, esi
        neg     eax
        and     eax, edi
        ret
alignDown32_Duncan(unsigned int, unsigned long):
        mov     ecx, esi
        mov     eax, -1
        sal     eax, cl
        and     eax, edi
        ret
```